Graphviz/DOT.sublime-syntax

```diff
@@ -65,56 +394,478 @@ contexts:
+
+###[ ATTRIBUTE LISTS ]#########################################################
+
+  attributes:
+    - include: fullline-comments-number-sign
+
+    - match: '[,;]'
+      scope: punctuation.separator.sequence.dot
+
+    - include: subgraph-declarations
+    - include: attribute-statements
+    - include: node-statements
+    - include: attribute-lists
+    - include: embedded-html
+    - include: literals
+    - include: operators
+
+  attribute-lists:
     - match: \[
-      scope: punctuation.definition.attributes.begin.dot
-      push:
-        - meta_scope: meta.attributes.dot
-        - match: \]
-          scope: punctuation.definition.attributes.end.dot
-          pop: true
-        - match: '[,;]'
-          scope: punctuation.separator.dot
-        - include: expressions
+      scope: punctuation.section.brackets.begin.dot
+      push: attribute-list-body
+    - include: stray-brackets
 
-  comments:
-    - match: //
-      scope: punctuation.definition.comment.dot
-      push:
-        - meta_scope: comment.line.double-slash.dot
-        - match: $\n?
-          pop: true
-    - match: '#'
-      scope: punctuation.definition.comment.dot
-      push:
-        - meta_scope: comment.line.number-sign.dot
-        - match: $\n?
-          pop: true
-    - match: /\*
-      scope: punctuation.definition.comment.dot
-      push:
-        - meta_scope: comment.block.dot
-        - match: \*/
-          scope: punctuation.definition.comment.dot
-          pop: true
-        - match: ^\s*(\*)(?!/)
-          captures:
-            1: punctuation.definition.comment.dot
+  attribute-list-body:
+    - meta_scope: meta.attributes.dot
+
+    - match: \]
+      scope: punctuation.section.brackets.end.dot
+      pop: 1
+
+    - include: attributes
+
+###[ PROTOTYPES ]##############################################################
+
+  eol-pop:
+    - match: '{{eol}}'
+      pop: 1
+
+###[ ILLEGALS ]################################################################
+
+  illegal-newline:
+    - match: $\n
+      scope: invalid.illegal.unexpected-newline.dot
+      pop: 1
+
+  stray-braces:
+    - match: \}
+      scope: invalid.illegal.stray.dot
+
+  stray-brackets:
+    - match: \]
+      scope: invalid.illegal.stray.dot
+
+###[ VARIABLES ]###############################################################
+
+variables:
+  leading_wspace: (?:^\s*)
+
+  eol: (?:${{line_break}}?)
+  line_break: (?:\n)
+
+  # https://graphviz.org/doc/info/lang.html#ids
+  id: |-
+    (?x:
+      [a-zA-Z_\x{200}-\x{377}][a-zA-Z0-9_\x{200}-\x{377}]*
+    )
+
+  edge: (?:edge|EDGE)
+  subgraph: (?:subgraph|SUBGRAPH)
+  graph: (?:graph|GRAPH)
+  digraph: (?:digraph|DIGRAPH)
+  digraph_or_graph: '(?:{{digraph}}|{{graph}})'
+  node: (?:node|NODE)
+  strict: (?:strict|STRICT)
+
+  declaration_keywords: |-
+    (?x:
+      {{edge}}
+    | {{node}}
+    )
+
+  compass_pt: |-
+    (?x:
+      n | ne | e | se | s | sw | w | nw | c | _
+    )
```

Comment:
Not important nor functional relevant, but a slightly more compact value style could probably also be used here.

```suggestion
  compass_pt: |-
    (?x: n | ne | e | se | s | sw | w | nw | c | _ )
```